code/modules/admin/antagonist_panel/antagonist_panel_tab.dm

```diff
@@ -0,0 +1,291 @@
+/**
+ *	Antagonist panel tab datums represent the individual tabs that may be present on an antagonist panel and are responsble
+ *	for generating the specific data contained within them.
+ */
+/datum/antagonist_panel_tab
+	/// The name of this tab to be displayed on the UI.
+	var/tab_name
+
+/// Returns data on the constituent sections of this tab, alongside the concomitant antagonist data.
+/datum/antagonist_panel_tab/proc/generate_section_data()
+	return
+
+/// Returns a list of areas and coordinates for each player entry in the tab, indexed by mind.
+/datum/antagonist_panel_tab/proc/generate_location_data()
+	return
+
+
+
+
+
+/datum/antagonist_panel_tab/generic
+	/// The antagonist ID that this tab should display data for.
+	var/antagonist_role
+
+/datum/antagonist_panel_tab/generic/New(tab_name, antagonist_role)
+	. = ..()
+
```

Comment:
```suggestion
```

Reply:
`/datum/antagonist_panel_tab/generic/New()` has code assigned to it, as does `/datum/antagonist_panel_tab/bundled/revolution/New()`, which you have also requested be removed.

Clarification on your requested changes would be appreciated.

```
/datum/antagonist_panel_tab/generic/New(tab_name, antagonist_role)
	. = ..()

	src.tab_name = tab_name
	src.antagonist_role = antagonist_role
```

Reply:
Oh, nevermind, whitespace removal.